ecosystem/sep-0024.md

```diff
@@ -190,6 +192,127 @@ The deposit flow can only be fulfilled if the Stellar `Account` has established
+1. At this point both the wallet and anchor are waiting on the trust line to be established in order to continue the deposit payment. This is described here [steps 4-6](#stellar-account-doesnt-trust-asset)
+
+##### Wallet Feature: Claimable Balance Polling
+
+Claimable Balance Polling allows the wallet to update the claimant that their is a balance available to claim and they may need to establish a [trust line](#stellar-account-doesnt-trust-asset) in order to do so.
+There are a few options as to how a wallet can poll and surface claimable balance information for their users.
+
+1. The wallet can poll Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/list/) list endpoint to pull any available claimable balances for the claimant.
+
+   * *Recommended request* `GET /claimable_balances?claimant={:claimant}&order={asc,desc}&limit={1-200}` or [See List Claimable Balances](https://developers.stellar.org/api/resources/claimablebalances/list/)
+
+2. The wallet can poll Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/single/) single endpoint to pull the submitted transactions' available claimable balance(s) for the claimant(user) using the claimable_balance_id.
+
+   * There are two recommended ways a wallet could extract the `claimable_balance_id` of the deposit transaction submitted
+
+   1. Wallets could pull the `claimable_balance_id` from anchor's transaction deposit [transaction(s) endpoint](#transaction-history) response
+
+   2. Wallets could decode the `results_xdr` of the deposit transaction submitted
+      1. Traverse each transaction's `results_xdr` to if it was submitted as a claimable balance
+      2. then pull the `claimable_balance_id` from the `xdr`
+      3. Use the `claimable_balance_id` to query Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/single/) single endpoint to pull all the information needed for the claimable balance.
+      4. Provide UI to showcase that their deposit was successful and that their's an available claimable balance as well as enable users to [claim a claimable balance.](#wallet-feature-claim-account-claimable-balance)
+
+      * Example of how a wallet could decode the `results_xdr`
+
+      ```js
+      const StellarBase = require('stellar-base');
+
+      const { xdr } = StellarBase;
+
+      const b64 = "AAAAAAAAAGQAAAAAAAAAAQAAAAAAAAAOAAAAAAAAAAA9/w3pAlg0fEoZp+eC9g0slbYtvoySN3i5y+F4WJY8sgAAAAA=";
+      const tr = xdr.TransactionResult.fromXDR(b64, 'base64');
+
+      const operationIndex = 0; // if you have multiple operations then you the index to correspond to he claimable balance operation
+      console.log(tr.result().results()[operationIndex].value().createClaimableBalanceResult().balanceId().toXDR('hex'));
+      ```
+
+##### Wallet Feature: Claim Account Claimable Balance
```

Comment:
I think it would be valuable to outline the operations needed right here. You can keep the docs reference if they want to check it out, but ideally they don't have to leave the SEP to know what the TX should look like. See SEP-10's description of challenge transactions for inspiration.

Reply:
I think I prefer to link both the documentation and showcase example code for example.
Similar to [SEP 20](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0020.md#example-code-for-linking)
Thoughts?

```suggestion
##### Wallet Feature: Claim Account Claimable Balance

It's recommended that Wallets provides a UI to enable accounts to claim their balances [polled](#wallet-feature-claimable-balance-polling)

[Example on how to claim a balance](https://developers.stellar.org/docs/glossary/claimable-balance/#claim-claimable-balance)
```javascript
const transaction = new TransactionBuilder(account, {
      fee: 100,
      networkPassphrase: this.network_passphrase,
    })
      .addOperation(
        Operation.claimClaimableBalance({
          balanceId,
        })
      )
      .setTimeout(0)
      .build()
    transaction.sign(keypair)
    this.logger.request(
      'Submitting claimClaimableBalance transaction',
      transaction
    )
    const result = await this.server.submitTransaction(transaction)
```
```

Reply:
That works, but note your current code is missing a change trust op. In order to accept a claimable balance the user needs to have a trustline to the asset. The tricky piece is that the change trust op only needs to be included once per asset. You should explain this nuance in this section before showing the code.